modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/OpcUaProtocolAdapter.java

```diff
@@ -194,11 +192,9 @@ public void stop(final @NotNull ProtocolAdapterStopInput input, final @NotNull P
 
     @Override
     public void discoverValues(
-            final @NotNull ProtocolAdapterDiscoveryInput input, final @NotNull ProtocolAdapterDiscoveryOutput output) {
-
-        if (opcUaClient == null) {
-            throw new IllegalStateException("OPC UA Adapter not started yet");
-        }
+            final @NotNull ProtocolAdapterDiscoveryInput input,
+            final @NotNull ProtocolAdapterDiscoveryOutput output) {
+        Objects.requireNonNull(opcUaClient, "OPC UA Adapter not started yet");
```

Comment:
not sure if this is really enough. After the check the client might get null.

Reply:
Fixed